qemu/tests/balloon_check.py

```diff
@@ -355,6 +356,94 @@ def get_win_mon_free_mem(self, session):
         else:
             self.test.fail("Failed to get windows guest free memory")
 
+    def run_balloon_service(self, session, win_drive_letter):
+        """
+        Start balloon service in windows guest
+
+        :param session: shell Object
+        :param win_drive_letter: virtio-win disk volume letter.
+        :return: cmd execution status and output
+        """
+        try:
+            start_cmd = self.params["run_balloon_service"] % win_drive_letter
+            status, output = session.cmd_status_output(start_cmd)
+            return (status, output)
+        except ShellCmdError:
+            self.test.error("Start balloon service failed!")
+
+    def check_balloon_service(self, session, win_drive_letter):
+        """
+        Check balloon service status in windows guest
+
+        :param session: shell Object
+        :param win_drive_letter: virtio-win disk volume letter.
+        :return: cmd execution status and output
+        """
+        try:
+            status_cmd = self.params["status_balloon_service"] % win_drive_letter
+            logging.debug("Check balloon service status.")
+            status, output = session.cmd_status_output(status_cmd)
+            return (status, output)
+        except ShellCmdError:
+            self.test.error("balloon service status check failed!")
+
+    def install_balloon_service(self, session, win_drive_letter):
+        """
+        Check/Install/Start balloon service for windows guest.
+
+        :param session: shell Object
+        :param win_drive_letter: virtio-win disk volume letter.
+        :return: cmd execution status and output
+        """
+        logging.debug("Check Balloon Service status in guest.")
+        status, output = self.check_balloon_service(session, win_drive_letter)
+        if re.search(r"running", output.lower(), re.M):
+            logging.debug("Balloon service is already running !")
+        elif re.search(r"stop", output.lower(), re.M):
+            logging.debug("Start Balloon Service in guest.")
+            try:
+                self.run_balloon_service(session, win_drive_letter)
+            except ShellCmdError:
+                raise exceptions.TestError("Start balloon service failed!")
```

Comment:
ShellCmdError is judeged in run_balloon_service, why judege it here?

Reply:
Will modify it, will use the returned status to rase test.error("Start balloon service failed!")